.github/workflows/main.yml

```diff
@@ -82,144 +86,99 @@ jobs:
-
-      - name: Upload artifact prep-build-dist-browserify-chrome
-        uses: actions/upload-artifact@v4
-        with:
-          name: prep-build-dist-browserify-chrome
-          path: ./dist/**/chrome # it's really in dist/chrome, but the ** is there so that it's uploaded relative to `dist`
-
-  prep-build-test-flask-browserify:
-    needs:
-      - prep-deps
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout and setup environment
-        uses: MetaMask/action-checkout-and-setup@v1
-        with:
-          is-high-risk-environment: false
-          skip-allow-scripts: true
-
-      - run: yarn build:test:flask
-
-      - name: Upload artifact prep-build-test-flask-browserify-chrome
-        uses: actions/upload-artifact@v4
-        with:
-          name: prep-build-test-flask-browserify-chrome
-          path: ./dist/**/chrome # it's really in dist/chrome, but the ** is there so that it's uploaded relative to `dist`
-
-  prep-build-test-flask-mv2-browserify:
-    needs:
-      - prep-deps
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout and setup environment
-        uses: MetaMask/action-checkout-and-setup@v1
-        with:
-          is-high-risk-environment: false
-          skip-allow-scripts: true
-
-      - run: yarn build:test:flask:mv2
-
-      - name: Upload artifact prep-build-test-flask-mv2-browserify-chrome
-        uses: actions/upload-artifact@v4
-        with:
-          name: prep-build-test-flask-mv2-browserify-chrome
-          path: ./dist/**/chrome # it's really in dist/chrome, but the ** is there so that it's uploaded relative to `dist`
-
-      - name: Upload artifact prep-build-test-flask-mv2-browserify-firefox
-        uses: actions/upload-artifact@v4
-        with:
-          name: prep-build-test-flask-mv2-browserify-firefox
-          path: ./dist/**/firefox # it's really in dist/firefox, but the ** is there so that it's uploaded relative to `dist`
+  build-dist-browserify:
+    uses: ./.github/workflows/run-build.yml
+    with:
+      build-name: build-dist-browserify
+      build-command: ${{ (github.head_ref || github.ref_name) == 'master' && 'yarn build prod' || 'yarn build dist' }}
+    secrets: inherit
+
+  build-dist-mv2-browserify:
+    uses: ./.github/workflows/run-build.yml
+    with:
+      build-name: build-dist-mv2-browserify
+      build-command: ${{ (github.head_ref || github.ref_name) == 'master' && 'yarn build prod' || 'yarn build dist' }}
+      mozilla-lint: true
+      enable-mv3: false
+    secrets: inherit
+
+  build-beta-browserify:
+    uses: ./.github/workflows/run-build.yml
+    with:
+      build-name: build-beta-browserify
+      build-command: yarn build --build-type beta dist
+    secrets: inherit
+
+  build-beta-mv2-browserify:
+    uses: ./.github/workflows/run-build.yml
+    with:
+      build-name: build-beta-mv2-browserify
+      build-command: yarn build --build-type beta dist
+      mozilla-lint: false # Disabled as it is failing for some reason
```

Comment:
Hmm 🤔 Not a big deal to disable this temporarily on some builds, as long as it remains enabled for `build-dist-mv2-browserify` (as if it fails there, it'll fail upon Firefox store submission as well).

Totally on board with resolving this in a follow-up PR, but I am curious to know why it fails. Is there a workflow run you could share that demonstrates this?